mypy/checkexpr.py

```diff
@@ -3938,20 +3938,15 @@ def is_valid_var_arg(self, typ: Type) -> bool:
     def is_valid_keyword_var_arg(self, typ: Type) -> bool:
         """Is a type valid as a **kwargs argument?"""
         if self.chk.options.python_version[0] >= 3:
-            return is_subtype(typ, self.chk.named_generic_type(
-                'typing.Mapping', [self.named_type('builtins.str'),
-                                   AnyType(TypeOfAny.special_form)]))
+            key_type = self.named_type('builtins.str')
         else:
-            return (
-                is_subtype(typ, self.chk.named_generic_type(
-                    'typing.Mapping',
-                    [self.named_type('builtins.str'),
-                     AnyType(TypeOfAny.special_form)]))
-                or
-                is_subtype(typ, self.chk.named_generic_type(
-                    'typing.Mapping',
-                    [self.named_type('builtins.unicode'),
-                     AnyType(TypeOfAny.special_form)])))
+            key_type = UnionType.make_union([
+                self.named_type('builtins.str'),
+                self.named_type('builtins.unicode')])
+        kwargs_type = self.chk.named_generic_type(
+            'typing.Mapping', [key_type, AnyType(TypeOfAny.special_form)])
+        kwargs_type.type.defn.type_vars[0].variance = COVARIANT
```

Comment:
Wouldn't this affect all uses of Mapping after this code runs?

Reply:
Ahh, you're right. Didn't realized that before.

Reply:
The fix I was thinking was something along these lines. 
```suggestion
        is_subtype(typ, self.chk.named_type('typing.Mapping')) and isinstance(typ.args[0], LiteralType)
```
This wouldn't cover the case of Literal["a", "b"] but you can try using the `try_getting_str_literals_from_type` method